sdk/storage/AzureStorageBlob/Source/TransferManager.swift

```diff
@@ -82,9 +81,68 @@ public protocol TransferDelegate: AnyObject {
 
 // MARK: Extensions
 
-internal extension TransferManager {
-    func transfer(withId id: UUID) -> Transfer? {
-        return transfers.first { $0.id == id }
+public extension Array where Element == Transfer {
+    /// Retrieve all upload transfers where the source matches the provided source URL.
+    /// - Parameters:
+    ///   - sourceURL: The URL to a file on this device.
+    func from(_ sourceURL: URL) -> [Transfer] {
```

Comment:
I also had this same thought :) However I didn't want to back us into a corner if for example, in the future we might want to include other kinds of transfers, for example perhaps MultiBlob transfers make sense to return here (or maybe not).

Customers interact with download by specifying a source blob, source container, and destination URL. They interact with upload by specifying a source URL, destination blob, and destination container. So I wanted to provide method to allow them to locate transfers matching either the source or destination, for both uploads and downloads.

Additionally, while it may not make sense, I don't think we actually guarantee that there is only ever one download for a given destination URL, or one upload for a given destination blob and container. So again I didn't want to back us into a corner here, and I wanted the interface for each of the methods to be similar. The only thing we can guarantee uniqueness on is the UUID of the transfer, so that's why it's the only method that returns a single (optional) transfer.

Reply:
Resolved.